flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/codegen/calls/ScalarOperatorGens.scala

```diff
@@ -186,7 +186,12 @@ object ScalarOperatorGens {
 
       case (TIME_WITHOUT_TIME_ZONE, INTERVAL_DAY_TIME) =>
         generateOperatorIfNotNull(ctx, new TimeType(), left, right) {
-          (l, r) => s"$l $op ((int) ($r))"
+          (l, r) => s"($l $op ((int) ($r))) % ${MILLIS_PER_DAY}"
```

Comment:
looks like similar lines can be added to `ScalarOperator.scala` class in flink-table-planner as well in order to support the test cases.